### ISSUE TITLE
metrics reconstruction from logs

### ISSUE THREAD

**Issue by itamar1992m** [ultralytics/ultralytics#8749]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Hi!
im trying to reconstruct the metrics from the logs after training and testing my data.

im using pretrained yolov8m and retrain it on my train set (and my val set)./
Then, im using the following code to test and examine the model on the testset:
        detections = self.model.val(project=output_path, name="results", split="test", save_json=True, mode='val')

in my project im detecting a specific custom labeland modify the coco 80 classes to have 81 labels where my custom label is the last one.

at first i tried to use a new coco labels file that contain only my new label :{0 "tbcow"} , but i observed a higher mAP for that specific label when using the 81 labels.

the only label that is relevant is the tbcow so i modified the confusion matrix to describe only that label with nc=2 (tbcow and background).
the problem is that i can't calculate the mAP for that new label after modifying the confusion matrix, and even when i tried to reconstruct the mAP for the "cow" and "tbcow" labels ,without modifying the confusion matrix, I didn't get the same mAP the showed in the following image describing my results on the test set:

<img width="586" alt="Screenshot 2024-03-06 182535" src="https://github.com/ultralytics/ultralytics/assets/157970621/236ca9f2-e886-4f1d-88ef-ffa11d16cece">

as you can see the tbcow (my custom label) is getting a high mAP (0.962) and the cow label which is very similar to the tbcow is getting a lower mAP (0.441) resulting a medium total avg mAP score (0.701).

My main goal is to analyze the accuracy on tbcow label ( i dont mind getting lower mAP on other labels).
So, how can i reconstruct or evaluate an accuracy score from the modified confusion matrix?
OR how can i extract the mAP of a specific label and not the total mAP from the log (in the image above) ?

In the experiment I did using only tbcow label I got the following confusion matrix and detection.results_dict:

i feel like (TP+TN)/(TP+TN+FP+FN) is not describing the accuracy well **~=0.86**.
the mAP i get from : detections.results_dict[''metrics/mAP50(B)'] is describing the accuracy better =0.94 .

<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/ITAMAR~1/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/ITAMAR~1/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">
<style>

</style>
</head>

<body link="#467886" vlink="#96607D">

metrics/precision(B) | metrics/recall(B) | metrics/mAP50(B) | metrics/mAP50-95(B) | fitness
-- | -- | -- | -- | --
0.928027 | 0.90712 | 0.946884 | 0.524597 | 0.566826

</body>

</html>

![confusion_matrix](https://github.com/ultralytics/ultralytics/assets/157970621/a6330940-4719-4d44-851f-2449566b539f)

How come they are so different?
